{"name": "测试数据：【 test004，123456，True，0，登录成功】", "status": "broken", "statusDetails": {"message": "KeyError: 'msg'", "trace": "self = <testcases.api_test.test_03_login.TestUserLogin object at 0x04FE4AF0>\nusername = 'test004', password = '123456', except_result = True, except_code = 0\nexcept_msg = '登录成功'\n\n    @allure.story(\"用例--登录用户\")\n    @allure.description(\"该用例是针对获取用户登录接口的测试\")\n    @allure.issue(\"https://www.cnblogs.com/wintest\", name=\"点击，跳转到对应BUG的链接地址\")\n    @allure.testcase(\"https://www.cnblogs.com/wintest\", name=\"点击，跳转到对应用例的链接地址\")\n    @allure.title(\"测试数据：【 {username}，{password}，{except_result}，{except_code}，{except_msg}】\")\n    @pytest.mark.single\n    @pytest.mark.parametrize(\"username, password, except_result, except_code, except_msg\",\n                             api_data[\"test_login_user\"])\n    def test_login_user(self, username, password, except_result, except_code, except_msg):\n        logger.info(\"*************** 开始执行用例 ***************\")\n>       result = login_user(username, password)\n\ntest_03_login.py:28: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _\n\nusername = 'test004', password = '123456'\n\n    def login_user(username, password):\n        \"\"\"\n        登录用户\n        :param username: 用户名\n        :param password: 密码\n        :return: 自定义的关键字返回结果 result\n        \"\"\"\n        result = ResultBase()\n        payload = {\n            \"username\": username,\n            \"password\": password\n        }\n        header = {\n            \"Content-Type\": \"application/json\"\n        }\n        res = user.login(data=json.dumps(payload), headers=header)\n        print(res.json())\n        result.success = False\n        if res.json()[\"code\"] == 200:\n            result.success = True\n            result.token = res.json()[\"login_info\"][\"data\"]\n            print(result.token)\n        else:\n>           result.error = \"接口返回码是 【 {} 】, 返回信息：{} \".format(res.json()[\"code\"], res.json()[\"msg\"])\nE           KeyError: 'msg'\n\nuser.py:99: KeyError"}, "description": "该用例是针对获取用户登录接口的测试", "attachments": [{"name": "log", "source": "751502e6-2115-42a4-9ed3-a6a6b46fa2fc-attachment.txt", "type": "text/plain"}, {"name": "stdout", "source": "99a9862b-d185-4a2a-8e4e-d3919254e586-attachment.txt", "type": "text/plain"}, {"name": "stderr", "source": "bd903fe5-6d38-4e71-8214-8528e3659088-attachment.txt", "type": "text/plain"}], "parameters": [{"name": "username", "value": "'test004'"}, {"name": "password", "value": "'123456'"}, {"name": "except_result", "value": "True"}, {"name": "except_code", "value": "0"}, {"name": "except_msg", "value": "'登录成功'"}], "start": 1622540420273, "stop": 1622540420303, "uuid": "ddd4248d-4b41-4b48-802a-5455773e43a7", "historyId": "7744887ae487c7b8afca75dd11cbf0e2", "testCaseId": "7272db244dabef390096479ae3196642", "fullName": "testcases.api_test.test_03_login.TestUserLogin#test_login_user", "labels": [{"name": "feature", "value": "用户登录模块"}, {"name": "epic", "value": "针对单个接口的测试"}, {"name": "severity", "value": "normal"}, {"name": "story", "value": "用例--登录用户"}, {"name": "tag", "value": "single"}, {"name": "parentSuite", "value": "testcases.api_test"}, {"name": "suite", "value": "test_03_login"}, {"name": "subSuite", "value": "TestUserLogin"}, {"name": "host", "value": "LAPTOP-1UC29CD2"}, {"name": "thread", "value": "12464-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "testcases.api_test.test_03_login"}], "links": [{"type": "test_case", "url": "https://www.cnblogs.com/wintest", "name": "点击，跳转到对应用例的链接地址"}, {"type": "issue", "url": "https://www.cnblogs.com/wintest", "name": "点击，跳转到对应BUG的链接地址"}]}